test/e2e/kvc_test.go

```diff
@@ -0,0 +1,370 @@
+		},
+		// Negative test cases.
+		{
+			description: "single vc - S3 - no label error",
+			volumeConfigs: []crv1.VolumeConfig{
+				{
+					ID:          "vol1",
+					Replicas:    1,
+					SourceType:  "S3",
+					SourceURL:   "s3://e2e-test/cifar-100-python.tar.gz",
+					EndpointURL: "http://10.63.255.0:9000",
+					AccessMode:  "ReadWriteOnce",
+					Capacity:    "5Gi",
+					Labels:      map[string]string{},
+					Options: map[string]string{
+						"awsCredentialsSecretName": "s3-creds",
+					},
+				},
+			},
+			expSuccess: false,
+			expError:   fmt.Sprintf("labels cannot be empty"),
+			expHP:      false,
+			expNA:      false,
+			expPVC:     false,
+		},
+		{
+			description: "single vc - S3 - no creds error",
+			volumeConfigs: []crv1.VolumeConfig{
+				{
+					ID:          "vol1",
+					Replicas:    1,
+					SourceType:  "S3",
+					SourceURL:   "s3://e2e-test/cifar-100-python.tar.gz",
+					EndpointURL: "http://10.63.255.0:9000",
+					AccessMode:  "ReadWriteOnce",
+					Capacity:    "5Gi",
+					Labels: map[string]string{
+						"key1": "val1",
+						"key2": "val2",
+					},
+					Options: map[string]string{},
+				},
+			},
+			expSuccess: false,
+			expError:   fmt.Sprintf("awsCredentialsSecretName key has to be set in options"),
+			expHP:      false,
+			expNA:      false,
+			expPVC:     false,
+		},
+		{
+			description: "single vc - NFS - no label error",
+			volumeConfigs: []crv1.VolumeConfig{
+				{
+					ID:         "vol1",
+					SourceType: "NFS",
+					AccessMode: "ReadWriteMany",
+					Capacity:   ".5Gi",
+					Labels:     map[string]string{},
+					Options: map[string]string{
+						"server": "10.63.249.182",
+						"path":   "/",
+					},
+				},
+			},
+			expSuccess: false,
+			expError:   fmt.Sprintf("labels cannot be empty"),
+			expHP:      false,
+			expNA:      false,
+			expPVC:     false,
+		},
+		{
+			description: "single vc - NFS - no server in options error",
+			volumeConfigs: []crv1.VolumeConfig{
+				{
+					ID:         "vol1",
+					SourceType: "NFS",
+					AccessMode: "ReadWriteMany",
+					Capacity:   ".5Gi",
+					Labels: map[string]string{
+						"key1": "val1",
+						"key2": "val2",
+					},
+					Options: map[string]string{
+						"path": "/",
+					},
+				},
+			},
+			expSuccess: false,
+			expError:   fmt.Sprintf("server has to be set in options"),
+			expHP:      false,
+			expNA:      false,
+			expPVC:     false,
+		},
+		{
+			description: "single vc - NFS - no path in options error",
+			volumeConfigs: []crv1.VolumeConfig{
+				{
+					ID:         "vol1",
+					SourceType: "NFS",
+					AccessMode: "ReadWriteMany",
+					Capacity:   ".5Gi",
+					Labels: map[string]string{
+						"key1": "val1",
+						"key2": "val2",
+					},
+					Options: map[string]string{
+						"server": "10.63.249.182",
+					},
+				},
+			},
+			expSuccess: false,
+			expError:   fmt.Sprintf("path has to be set in options"),
+			expHP:      false,
+			expNA:      false,
+			expPVC:     false,
+		},
```

Comment:
I think these negative tests are also tested in the unit tests. The one's we should negative test here are the cluster level things like:
1. minio endpoint is not reachable
2. s3 path is incorrect
3. nfs server ip is unreachable
4. nfs path is not correct.
5. One volumeConfig passes while the other one fails.

Maybe they can go in a follow-up PR too and we can just add a TODO here.

Reply:
Ok. I will work on adding these cases.

Reply:
I've added all the above test cases.